enterprise/server/scheduling/scheduler_client/scheduler_client.go

```diff
@@ -180,6 +187,8 @@ func (r *Registration) ServeStatusz(w http.ResponseWriter, req *http.Request) {
 }
 
 func (r *Registration) processWorkStream(ctx context.Context, stream scpb.Scheduler_RegisterAndStreamWorkClient, schedulerMsgs chan *scpb.RegisterAndStreamWorkResponse, schedulerErr chan error, registrationTicker, requestMoreWorkTicker *time.Ticker) (bool, error) {
+	// Update queue length before creating registration message
+	r.node.CurrentQueueLength = int32(r.taskScheduler.QueueLength())
```

Comment:
this code is pretty brutal, sorry. Is the race detector kosher with this?

Reply:
agree it feels gross - would it feel better if the queue length wasn't part of the node proto itself and it was just part of the outer RegisterAndStreamWorkRequest? that way we're not modifying the node all the time?

Reply:
tried going down that road a bit too, but also felt gnarly in a different way

re: race detector - i tried running the executor with `--config=race` (on mac) get crashes with or without this change lol. not sure if i'm doing it wrong though

open to ideas on making this better, also down with scrapping this altogether

Reply:
how bout instead of mutating r.node and passing that value back in the RegisterAndStreamWorkRequest -- we proto.Clone(r.node) here and just slap the length on the local copy? 

I think you could also revert the changes to makeExecutionNode because that's only called on setup when the queue length is going to be 0 anyway?

Reply:
also to be clear I meant this code is brutal, not your additions to it! there's a lot of confusing goroutine stuff in here that I find pretty tough to follow.